cli/src/commands/config/mod.rs

```diff
@@ -67,8 +66,13 @@ impl ConfigLevelArgs {
         }
     }
 
-    fn config_paths<'a>(&self, config_env: &'a ConfigEnv) -> Result<Vec<&'a Path>, CommandError> {
+    fn config_paths<'a>(
+        &self,
+        ui: &Ui,
+        config_env: &'a ConfigEnv,
+    ) -> Result<Vec<&'a ConfigPath>, CommandError> {
         if self.user {
+            config_env.warn_for_deprecated_paths(ui);
```

Comment:
Does this mean the warning won't be printed unless the user tried to edit/set/list/get the config file? That would be bad because the legacy config file would silently become ignored.

As I said, I would make `ConfigEnv::from_environment()` emit the warning because that's simpler and we'll remove it later.

Reply:
> The idea of making this warning not respect how the user set up their stuff feels very wrong to me...

What do you mean by "not respect how the user set up their stuff"?

`ConfigEnv::from_environment(ui)` -> `UnresolvedConfigEnv::resolve(ui)` look up config paths, so it can emit the warning only when a file exists at the deprecated location.

Do you mean it's wrong to not respect `--color`, etc.? I agree, but I think that's acceptable given this warning is temporary, and the user will have to react immediately to suppress the warning.